File: hono-resend-examples/javascript/examples/inbound.js
import "dotenv/config";
import { Resend } from "resend";

const resend = new Resend(process.env.RESEND_API_KEY);

// This example fetches a previously received inbound email by ID.
// The email ID comes from an "email.received" webhook event payload.
const emailId = process.env.INBOUND_EMAIL_ID || "example-email-id";

if (emailId === "example-email-id") {
  console.log("Note: Set INBOUND_EMAIL_ID to fetch a real inbound email.");
  console.log("You get this ID from the 'email.received' webhook event.\n");
}

const { data: email, error } = await resend.emails.get(emailId);

if (error) {
  console.error("Error fetching email:", error);
  process.exit(1);
}

console.log("=== Inbound Email Details ===");
console.log("From:", email?.from);
console.log("To:", email?.to);
console.log("Subject:", email?.subject);
console.log("Created:", email?.created_at);

if (email?.text) {
  const preview = email.text.length > 200 ? email.text.slice(0, 200) + "..." : email.text;
  console.log("\nText preview:\n" + preview);
}

console.log("\nDone!");

File: rust-resend-examples/examples/inbound.rs
use resend_rs::Resend;

#[tokio::main]
async fn main() {
    dotenvy::dotenv().ok();

    let api_key = std::env::var("RESEND_API_KEY").expect("RESEND_API_KEY environment variable is required");

    let resend = Resend::new(&api_key);

    // This example fetches a previously received inbound email by ID.
    // The email ID comes from an "email.received" webhook event payload.
    // Set up inbound webhooks at: https://resend.com/webhooks
    let email_id = std::env::var("INBOUND_EMAIL_ID").unwrap_or_else(|_| "example-email-id".to_string());

    if email_id == "example-email-id" {
        println!("Note: Set INBOUND_EMAIL_ID to fetch a real inbound email.");
        println!("You get this ID from the 'email.received' webhook event.\n");
    }

    match resend.emails.get(&email_id).await {
        Ok(email) => {
            println!("=== Inbound Email Details ===");
            println!("From: {}", email.from);
            println!("To: {:?}", email.to);
            println!("Subject: {}", email.subject);
            println!("Created: {}", email.created_at);

            if let Some(text) = &email.text {
                let preview = if text.len() > 200 {
                    format!("{}...", &text[..200])
                } else {
                    text.clone()
                };
                println!("\nText preview:\n{}", preview);
            }

            println!("\nDone!");
        }
        Err(e) => {
            eprintln!("Error fetching email: {}", e);
            std::process::exit(1);
        }
    }
}
